src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsdocumentmanagementapi/config/DocumentRequestContextConfiguration.kt

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.digital.hmpps.hmppsdocumentmanagementapi.config
+
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.slf4j.LoggerFactory
+import org.springframework.context.annotation.Configuration
+import org.springframework.web.servlet.HandlerInterceptor
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
+import uk.gov.justice.digital.hmpps.hmppsdocumentmanagementapi.resource.SERVICE_NAME
+import uk.gov.justice.digital.hmpps.hmppsdocumentmanagementapi.resource.USERNAME
+
+@Configuration
+class ClientTrackingConfiguration(private val documentRequestContextInterceptor: DocumentRequestContextInterceptor) : WebMvcConfigurer {
+  override fun addInterceptors(registry: InterceptorRegistry) {
+    log.info("Adding document request context interceptor")
+    registry.addInterceptor(documentRequestContextInterceptor).addPathPatterns("/documents/**")
+  }
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+}
+
+@Configuration
+class DocumentRequestContextInterceptor : HandlerInterceptor {
+  override fun preHandle(request: HttpServletRequest, response: HttpServletResponse, handler: Any): Boolean {
+    val serviceName = request.getHeader(SERVICE_NAME)?.trim()
```

Comment:
Encapsulating the logic to retrieve the service name and username values here means it can be switched from using headers to extracting from tokens if that's the preferred approach.